src/Hpack/Config.hs

```diff
@@ -603,6 +674,20 @@ expandCSources dir sect@Section{..} = do
   (warnings, files) <- expandGlobs "c-sources" dir sectionCSources
   return (warnings, sect {sectionCSources = files})
 
+toCustomSetup :: Maybe BuildType -> CustomSetupSection -> ([String], CustomSetup)
+toCustomSetup mBuildType CustomSetupSection{..} = (warnings, setup)
+  where
+    warnings :: [String]
+    warnings =
+      case mBuildType of
+        Nothing -> []
+        Just Custom -> []
+        Just buildType -> ["Existence of `custom-setup` stanza implies `build-type: Custom`, but package.yaml sets `build-type: " ++ show buildType ++ "`"]
```

Comment:
I wouldn't even care about this and leave it to `cabal` the case when a user specifies nonsense here.

Are you comfortable with removing this warning?